src/modules/EventModule.cpp

```diff
@@ -0,0 +1,463 @@
+
+		if (args.size() != 1)
+		{
+			client.createMessage(message.channelId(), "**Wrong Usage of Command!** ");
+			return;
+		}
+		if (config.isEventRunning == false)
+		{
+			client.createMessage(message.channelId(), "**No event is active at this moment.**");
+			return;
+		}
+		if (config.eventHighRiskHighRewardRunning)
+		{
+			int num = config.eventTimer->remainingTime();
+			QString time = utility::StringifyMilliseconds(num);
+
+			Discord::Embed embed;
+			embed.setColor(15844367);
+			embed.setTitle("**HighRiskHighReward** event");
+			embed.setDescription("The steal chance is **decreased** but if you succeed you will get **bonus " + currencyConfig.currencySymbol + "**!\n"
+				"Event **expires** in `" + time + "`");
+
+			client.createMessage(message.channelId(), embed);
+			return;
+
+		}
+		if (config.eventLowRiskLowRewardRunning)
+		{
+			int num = config.eventTimer->remainingTime();
+			QString time = utility::StringifyMilliseconds(num);
+
+			Discord::Embed embed;
+			embed.setColor(15844367);
+			embed.setTitle("**LowRiskLowReward** event");
+			embed.setDescription("The steal chance is **increased** but you have to give a **penalty** if you succeed!\n"
+				"Event **expires** in `" + time + "`");
+
+			client.createMessage(message.channelId(), embed);
+			return;
+		}
+
+	});
+	RegisterCommand(Commands::EVENT_GIVE_NEW_ACCESS, "give-new-event-access", [this](Discord::Client& client, const Discord::Message& message, const Discord::Channel& channel)
+		{
+			QStringList args = message.content().split(' ');
+
+			Permissions::ContainsPermission(client, channel.guildId(), message.author().id(), CommandPermission::ADMIN, [this, args, message, channel](bool result)
+				{
+					GuildSetting* setting = &GuildSettings::GetGuildSetting(channel.guildId());
+					if (!result)
+					{
+						UmikoBot::Instance().createMessage(message.channelId(), "**You don't have permissions to use this command.**");
+						return;
+					}
+					if (args.size() < 2)
+					{
+						UmikoBot::Instance().createMessage(channel.id(), "**Wrong Usage of Command!**");
+						return;
+					}
+					snowflake_t guild = channel.guildId();
+					snowflake_t chan = channel.id();
+
+					auto& roles = message.mentionRoles();
+
+					if (roles.isEmpty())
+					{
+						UmikoBot::Instance().createMessage(channel.id(), "**Wrong Usage of Command!\n**Please mention some role(s) so that I can add them.");
+						return;
+					}
+					for (auto& role : roles)
+					{
+						if (m_EventWhitelist[guild].contains(role)) continue;
+						m_EventWhitelist[guild].push_back(role);
+					}
+
+					UmikoBot::Instance().createMessage(channel.id(), "**The roles have been added.**\nPeople with the role(s) can now launch & end events!");
+
+				});
+		});
+	RegisterCommand(Commands::EVENT_TAKE_NEW_ACCESS, "take-new-event-access", [this](Discord::Client& client, const Discord::Message& message, const Discord::Channel& channel)
```

Comment:
Take *new* event access?

Reply:
Yes, take event launching and ending access from the people who have the access! (Works like the polls)